python/ray/tune/checkpoint_manager.py

```diff
@@ -198,7 +202,9 @@ def best_checkpoints(self):
     def _priority(self, checkpoint):
         result = flatten_dict(checkpoint.result)
         priority = result[self._checkpoint_score_attr]
-        return -priority if self._checkpoint_score_desc else priority
+        if self._checkpoint_score_desc:
+            priority = -priority
+        return (not is_nan(priority), priority)
```

Comment:
If we're changing this to a tuple anyway, let's include the `checkpoint.order`, too - that way the latest nan checkpoints are preserved (e.g. if the metric is always nan)
```suggestion
        return (not is_nan(priority), priority, checkpoint.order)
```

Reply:
Thanks, done